sdk/communication/communication-rooms/src/roomsClient.ts

```diff
@@ -157,25 +157,10 @@ export class RoomsClient {
     pageSettings: ListPageSettings,
     options: ListRoomOptions = {},
   ): AsyncIterableIterator<CommunicationRoom[]> {
-    if (!pageSettings.continuationToken) {
-      const currentSetResponse = await this.client.rooms.list(options);
-      pageSettings.continuationToken = currentSetResponse.nextLink;
-      if (currentSetResponse.value) {
-        yield currentSetResponse.value.map((room) => mapCommunicationRoomToSDKModel(room));
-      }
-    }
-
-    while (pageSettings.continuationToken) {
-      const currentSetResponse = await this.client.rooms.listNext(
-        pageSettings.continuationToken,
-        options,
-      );
-      pageSettings.continuationToken = currentSetResponse.nextLink;
-      if (currentSetResponse.value) {
-        yield currentSetResponse.value.map((room) => mapCommunicationRoomToSDKModel(room));
-      } else {
-        break;
-      }
+    const currentSetResponse = await this.client.rooms.list(options);
```

Comment:
This section was changed , as the autogen changed the signature of list method from Promise<ParticipantsListResponse> to PagedAsyncIterableIterator<RoomParticipant>. We previously had to iterate with the next link and now with the PagedAsyncIterableIterator we can use the for await